enrollment.md

```diff
@@ -0,0 +1,153 @@
+# WEBCAT Enrollment Specification
+
+This document explains how domains enroll into WEBCAT and how validators and oracles validate enrollment requests.
+
+The enrollment data are processed and stored in a permissioned CometBFT chain, where the voting network consists of trusted partner orgs (other nonprofits, Tor relay associations, browsers) acting as validators in the CometBFT consensus. Any third party can operate an observer node that allows them to check consistency, behavior, or monitor entries and updates.
+
+# Background
+
+[CometBFT](https://docs.cometbft.com/v1.0/) is a Byzantine Fault Tolerant (BFT) consensus algorithm derived from [Tendermint](https://arxiv.org/abs/1807.04938). It provides a round-based protocol that will reach consensus as long as at least $\gt 2/3$ of the validator voting power is online and honest. The benefit of using CometBFT is that consensus and the networking parts of the chain are handled by an external dependency.
+
+The blockchain state machine runs as an [ABCI](https://docs.cometbft.com/v1.0/) application. The consensus algorithm orders transactions into blocks, and the ABCI application executes them into the application state.
+
+The ABCI interface consists of:
+- `CheckTx`: basic validity checks run on transactions prior to them being added to the mempool
+- `DeliverTx`: validity checks that run when the transaction is included in a block
+- `BeginBlock`/`EndBlock`: hooks to run logic that runs at the beginning and end of a block boundary respectively
+- `Commit`: finalizes the application state for that block
+
+After executing all transactions in a block, the Merkleized application state root, the AppHash, is committed into the block header and signed by validators. This will be used by WEBCAT as a verifiable timestamped commitment.
+
+## Timestamp
+
+The CometBFT AppHash is a timestamped commitment, so it can serve as the timestamp to ensure that manifests cannot be signed with future dates.
+
+Each block produced by the enrollment network has a block height (a monatonic counter enforced by consensus), the corresponding AppHash, and a consensus timestamp. By signing a manifest that includes `(AppHash, height, timestamp)`, the manifest signer demonstrates that the manifest could not have existed before that block was finalized.
+
+Previous proofs (older AppHash values) can be reused in manifests, allowing for backdating. This is acceptable because the security goal is to prevent future-dating, wherein a manifest would be valid longer than it should be.
+
+The chain itself does not enforce expiry; clients check the expiry using the consensus timestamp.
+
+The consensus data will be published to a CDN with the following structure:
+
+```json
+{
+  "height": 12345,
+  "app_hash": "abc123def456...",
+  "timestamp": 1640995200,
+  "block_time": "2025-05-01T00:00:00Z"
+}
+```
+
+This allows clients to verify the `AppHash` against the chain and check manifest expiry using the consensus timestamp.
```

Comment:
Here, I would leave it to manifests to contain the whole set of information needed to verify a timestamp. Otherwise, the browser need to fetch elsewhere the consensus information for a specific apphash?

Reply:
updated with 826575546a6a5aed32a5761475de84e0136474f3 based on discussion - every day we publish to the CDN the latest LightBlock, it has 1) the Commit which has a list of signatures, and 2) the Header which includes the AppHash and the timestamp
https://github.com/cometbft/cometbft/blob/v0.34.x/spec/core/data_structures.md#lightblock